### DISCUSSION TITLE
How do we deploy multiple functions?

### DISCUSSION THREAD

**Discussion by HeavenOSK** [GoogleCloudPlatform/functions-framework-dart#311] [Q&A] [answered]
Nice to meet you, I'm new to GCP and this may be a basic question.
 
# Overview
I want to deploy two different functions for GCP. But I could not deploy them. 
How do we add multiple functions?
 
My code is as followings: 
```dart
import 'package:functions_framework/functions_framework.dart';
import 'package:shelf/shelf.dart';

@CloudFunction(target: 'hello-world')
Response helloWorld(Request request) {
  return Response.ok('Hello, world!');
}

@CloudFunction(target: 'next-world')
Response nextWorld(Request request) {
  return Response.ok('Next World!');
}
```

And, I use the following command.

```shell
$ gcloud beta run deploy hello --allow-unauthenticated --source=.   
```
 
 Thanks.

**Comment by kevmoo** [comment, marked as answer]
We probably need to surface this better – https://github.com/GoogleCloudPlatform/functions-framework-dart/blob/main/docs/quickstarts/01-quickstart-dart.md#changing-the-default-function-name

`function` is the default, but if you want to override it you need to set the `FUNCTION_TARGET` environment variable when you deploy!
